middlewares/accesslog/logger_test.go

```diff
@@ -68,10 +69,55 @@ func TestLogger(t *testing.T) {
 
 	logger.ServeHTTP(&logtestResponseWriter{}, req, LogWriterTestHandlerFunc)
 
-	if logdata, err := ioutil.ReadFile(logfilePath); err != nil {
+	if logdata, err := ioutil.ReadFile(logfilePath); err == nil {
+		assertValidLogData(t, logdata)
+	} else {
 		fmt.Printf("%s\n%s\n", string(logdata), err.Error())
 		assert.NoError(t, err)
-	} else if tokens, err := shellwords.Parse(string(logdata)); err != nil {
+	}
+}
+
+func TestNewLogHandlerOutputStdout(t *testing.T) {
+	file, err := ioutil.TempFile("", "testlogger")
+	if err != nil {
+		t.Fatalf("failed to create temp file: %s", err)
+	}
+	defer file.Close()
+	// capture stdout in a file
+	stdOut := os.Stdout
+	os.Stdout = file
+
+	logger, err := NewLogHandler("")
+	require.NoError(t, err)
+	defer logger.Close()
+
+	req := &http.Request{
+		Header: map[string][]string{
+			"User-Agent": {testUserAgent},
+			"Referer":    {testReferer},
+		},
+		Proto:      testProto,
+		Host:       testHostname,
+		Method:     testMethod,
+		RemoteAddr: fmt.Sprintf("%s:%d", testHostname, testPort),
+		URL: &url.URL{
+			User: url.UserPassword(testUsername, ""),
+			Path: testPath,
+		},
+	}
+
+	logger.ServeHTTP(&logtestResponseWriter{}, req, LogWriterTestHandlerFunc)
+
+	written, err := ioutil.ReadFile(file.Name())
```

Comment:
Good catch, I was a bit surprised by the compiler does not complain even though we do not use err anymore after this call. The reason is clear though, its already initialised and used before.